packages/react-celo/src/hooks/use-providers.ts

```diff
@@ -29,21 +28,37 @@ export function walletToProvider(wallet: WalletEntry): WalletConnectProvider {
   };
 }
 
-export function getRecent(): Maybe<Provider> {
-  const type = getTypedStorageKey(localStorageKeys.lastUsedWalletType);
-  const id = getTypedStorageKey(localStorageKeys.lastUsedWalletId);
-  let provider: Maybe<Provider>;
+export function getRecent(): { ids: Set<string>; providers: Provider[] } {
+  const listOfRecentWalletTypeIds = getRecentWallets();
 
-  if (id && WalletTypes.WalletConnect === type) {
-    provider = Object.values(PROVIDERS).find(
-      (p) => (p as WalletConnectProvider).walletConnectId === id
-    );
-  } else {
-    provider = Object.values(PROVIDERS).find((p) => p.type === type);
-  }
+  // create mapping by the identifier we saved so its easy to get the recent provider from the id
+  const providersByTypeId = Object.values(PROVIDERS).reduce(
+    (collection, current) => {
+      const unifiedID =
+        current.type === WalletTypes.WalletConnect
+          ? `${(current as WalletConnectProvider).type}:${
+              (current as WalletConnectProvider).walletConnectId
```

Comment:
nitpick, why the concatenation? It could be just `walletConnectId`, right?

Reply:
i guess so?